sdk/messaging/azeventhubs/internal/rpc.go

```diff
@@ -98,6 +100,17 @@ type RPCLinkArgs struct {
 	LogEvent azlog.Event
 }
 
+func closeOrLog(name string, limit time.Duration, closeable interface {
+	Close(ctx context.Context) error
+}) {
+	ctx, cancel := context.WithTimeout(context.Background(), limit)
+	defer cancel()
+
+	if err := closeable.Close(ctx); err != nil {
```

Comment:
What's the real-world impact here if `Close()` fails?  Does anything leak?

Reply:
It's _possible_ something leaks. I had an odd error where EH was complaining that I had the $cbs link opened twice and this was the only case I could think of where that might be.